### PR TITLE
Add tags and properties parameters to upload* functions

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: api/src/main/scala/dx/api/DxApi.scala [hunk 1]
```diff
@@ -892,17 +892,26 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DXEnvironment
...
       try {
         // shell out to dx upload. We need to quote the path, because it may contain spaces
-        val destOpt = destination.map(d => s"""--destination "${d}" -p""").getOrElse("")
-        val waitOpt = if (wait) "--wait" else ""
-        val dxUploadCmd = s"""dx upload "${path.toString}" --brief ${destOpt} ${waitOpt}"""
+        val destOpt = destination.map(d => s""" --destination "${d}" -p""").getOrElse("")
+        val waitOpt = if (wait) " --wait" else ""
+        val tagsOpt = tags.map(tag => s" --tag ${tag}").mkString("")
+        val propertiesOpt = properties.map {
+          case (key, value) => s" --property ${key}=${value}".mkString("")
```
Comment (r-i-v-a) [inline api/src/main/scala/dx/api/DxApi.scala:911]: This map() puts out a collection, so the final command string is going to have something like "List( --property a=A,  --property b=B)" in it

I've tried this in a sandbox project 
```
	val m = Map("a" -> "A", "b" -> "B")

	val s = m.map {
		case (key, value) => s" --property ${key}=${value}".mkString("")
	}

	val s1 = s"""start ${s} end""" 

	println(s1)
```

Comment (r-i-v-a) [reply api/src/main/scala/dx/api/DxApi.scala:911]: @jdidion   Whereas you probably want to do
```
	val m = Map("a" -> "A", "b" -> "B")

	val s = m.map {
		case (key, value) => s" --property ${key}=${value}"
	}

	val s1 = s"""start ${s.mkString("")} end""" 

	println(s1)
```

**Review by r-i-v-a — CHANGES_REQUESTED** [review]
Fix joining collections into command string for tags, properties

File: api/src/main/scala/dx/api/DxApi.scala (hunk 1 above)
Comment (jdidion) [reply api/src/main/scala/dx/api/DxApi.scala]: yep good catch

**Review by r-i-v-a — APPROVED** [review]

**Pull request merged**